src/logit_logger.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * logit_logger.cpp
+ *
+ *  Created on: May 20, 2021
+ *      Author: pnikiel
+ */
+
+#include <logit_logger.h>
+#include <LogIt.h>
+
+static Log::LogComponentHandle logItComponentHandle = Log::INVALID_HANDLE;
+
+void initializeOpen62541LogIt (Log::LOG_LEVEL logLevel)
+{
+	// register LogIt component for open62541
+	Log::LogComponentHandle handle = Log::getComponentHandle("open62541");
+	if (handle == Log::INVALID_HANDLE)
+	{
+		logItComponentHandle = Log::registerLoggingComponent("open62541", logLevel);
+	}
+}
+
+static void logFromOpen62541 (
+		void *logContext,
+		UA_LogLevel level,
+		UA_LogCategory category,
+        const char *msg,
+		va_list args)
+{
+	// translate open62541 log level into LogIt log level
+	Log::LOG_LEVEL logItLogLevel;
+	switch (level)
+	{
+		case UA_LOGLEVEL_DEBUG: logItLogLevel = Log::DBG; break;
+		case UA_LOGLEVEL_ERROR: logItLogLevel = Log::ERR; break;
+		case UA_LOGLEVEL_FATAL: logItLogLevel = Log::ERR; break;
+		case UA_LOGLEVEL_INFO: logItLogLevel = Log::INF; break;
+		case UA_LOGLEVEL_TRACE: logItLogLevel = Log::TRC; break;
+		case UA_LOGLEVEL_WARNING: logItLogLevel = Log::WRN; break;
+		default: logItLogLevel = Log::ERR; // just in case they added a new level.
+	}
+	// translate open62541 category.
```

Comment:
optimisation thing... at this point in the code it might be worth checking to see if the open62541 component is logging at this verbosity by calling the LogIt fn below
`SHARED_LIB_EXPORT_DEFN bool isLoggable(const Log::LOG_LEVEL& level, const LogComponentHandle& componentHandle);`

If not then you can just exit the method, since the eventual call to the LOG macro will evaluate to noop anyway. Might save some string comparisons in the category switch below